driver/others/memory.c

```diff
@@ -415,23 +417,72 @@ int openblas_get_num_threads(void) {
 #endif
 }
 
-struct release_t {
-  void *address;
-  void (*func)(struct release_t *);
-  long attr;
-};
-
 int hugetlb_allocated = 0;
 
 #if defined(OS_WINDOWS)
 #define THREAD_LOCAL __declspec(thread)
-#define UNLIKELY_TO_BE_ZERO(x) (x)
+#define LIKELY_ONE(x) (x)
 #else
 #define THREAD_LOCAL __thread
-#define UNLIKELY_TO_BE_ZERO(x) (__builtin_expect(x, 0))
+#define LIKELY_ONE(x) (__builtin_expect(x, 1))
+#endif
+
+struct alloc_t {
+  int used;
+  int attr;
+  void (*func)(struct alloc_t *);
+  /* Pad to 64-byte alignment */
+#ifdef __64BIT__
+  char pad[48];
+#else
+  char pad[52];
+#endif
+};
+
+static const int allocation_block_size = BUFFER_SIZE + sizeof(struct alloc_t);
+
+/* TLS is supported from clang 2.8, gcc 4.1, MSVC 2005, and XCode 8 */
```

Comment:
this is a rather convoluted test, perhaps break this into multiple more easily checkable/modifiable define statements?

Reply:
along the lines of the android test you have below, which is considerably easier to read.